test/DGmethods/Horizontal_integral_test.jl

```diff
@@ -0,0 +1,313 @@
+      
+      for e in 1:nrealelem
+        for k in 1:Nqk
+          for j in 1:Nq
+            for i in 1:Nq
+              ijk = i + Nq * ((j-1)+ Nq * (k-1))
+              S[k] += localvgeo[ijk,grid.x1id,e] * localvgeo[ijk,grid.MHid,e]
+              S1[k] += localvgeo[ijk,grid.MHid,e]
+            end
+          end
+        end
+      end
+      
+      Stot = zeros(Nqk)
+      S1tot = zeros(Nqk)
+      Err = 0
+      
+      for k in 1:Nqk
+        Stot[k] = MPI.Reduce(S[k], +, 0, MPI.COMM_WORLD)
+        S1tot[k] = MPI.Reduce(S1[k], +, 0, MPI.COMM_WORLD)
+        Err += (0.5 - Stot[k] / S1tot[k])^2
+      end
+      Err = sqrt(Err / Nqk)
+      @test Err< 2e-15
+    
+  end
+
+  warpfun1 = (ξ1, ξ2, ξ3) -> begin
+    x1 = sin(2 * π * ξ3)/16 + ξ1 
+    x2 = ξ2  + (ξ2 - 1/2) * cos(2 * π * ξ2 * ξ3) / 4
+    x3 = ξ3 + sin(2π * (ξ1))/20
+    return (x1, x2, x3)
+  end
+
+  for N in 4:4
+    Ne = 1 
+      brickrange1 = (range(FT(0); length=Ne+1, stop=1),
+              range(FT(0); length=Ne+1, stop=1),
+              range(FT(0); length=2, stop=1))
+      topl1 = StackedBrickTopology(mpicomm, brickrange1,
+                            periodicity = (false, false, false))
+
+      grid1 = DiscontinuousSpectralElementGrid(topl1,
+                                        FloatType = FT,
+                                        DeviceArray = ArrayType,
+                                        polynomialorder = N,
+                                        meshwarp = warpfun1,
+                                       )
+
+      N = polynomialorder(grid1)
+      vgeo1 = grid1.vgeo
+      Nq = N + 1
+      Nqk = dimensionality(grid1) == 2 ? 1 : Nq
+      nrealelem = length(topl1.realelems)
+      host_array = Array ∈ typeof(vgeo1).parameters
+      localvgeo = host_array ? vgeo1 : Array(vgeo1)
+      S = zeros(Nqk)
+      S1 = zeros(Nqk)
+      K = zeros(Nqk)
+      for e in 1:nrealelem
+        for k in 1:Nqk
+          for j in 1:Nq
+            for i in 1:Nq
+              ijk = i + Nq * ((j-1)+ Nq * (k-1))
+              S[k] += localvgeo[ijk,grid1.x1id,e] * localvgeo[ijk,grid1.MHid,e]
+              S1[k] += localvgeo[ijk,grid1.MHid,e]
+	      K[k] = localvgeo[ijk,grid1.x3id,e]
+             end
+          end
+        end
+      end
+
+      Stot = zeros(Nqk)
+      S1tot = zeros(Nqk)
+      Err = 0
+
+      for k in 1:Nqk
+        Stot[k] = MPI.Reduce(S[k], +, 0, MPI.COMM_WORLD)
+        S1tot[k] = MPI.Reduce(S1[k], +, 0, MPI.COMM_WORLD)
+        Err += (0.5 + sin(2 * π * K[k]) / 16 - Stot[k] / S1tot[k])^2
+	end
+
+      Err = sqrt(Err / Nqk)
+      @test 2e-15 > Err
+    
+  end
+
+  N = 4
+  base_Nhorz = 4
+  base_Nvert = 2
+  Rinner = 1 // 2  
+  Router = 1
+  expected_result = [-4.5894269717905445e-8 -1.1473566985387151e-8;
+		     -2.0621904184281448e-10  -5.155431637149377e-11;
+	             -8.72191208145523e-13 -2.1715962361668062e-13] 
+  for l in 1:3
+      Nhorz = 2^(l-1) * base_Nhorz
+      Nvert = 2^(l-1) * base_Nvert
```

Comment:
If you do a fixed number of vertical levels, you could check the error in all the slabs of the mesh (not just the inner and outer)